packages/components/src/button/index.js

```diff
@@ -15,9 +19,85 @@ import { forwardRef } from '@wordpress/element';
  */
 import Tooltip from '../tooltip';
 import Icon from '../icon';
+import * as buttonStyles from './styles';
 
 const disabledEventsOnDisabledButton = [ 'onMouseDown', 'onClick' ];
 
+const getBaseStyles = ( props ) => {
+	if ( props.isPrimary ) {
+		return buttonStyles.primary.styles;
+	}
+
+	if ( props.isSecondary ) {
+		return buttonStyles.secondary.styles;
+	}
+
+	if ( props.isTertiary ) {
+		return buttonStyles.tertiary.styles;
+	}
+
+	if ( props.isLink ) {
+		if ( props.isDestructive ) {
+			return css`
+				${ buttonStyles.link.styles }
+				${ buttonStyles.link.destructive }
+			`;
+		}
+		return buttonStyles.link.styles;
+	}
+
+	// check pure destructive last to allow for destructive links
+	if ( props.isDestructive ) {
+		return buttonStyles.destructive.styles;
+	}
+
+	return buttonStyles.shared.buttonBase;
+};
+
+const getBusyStyles = ( props ) => {
+	if ( ! props.isBusy ) {
+		return '';
+	}
+
+	if ( props.isPrimary ) {
+		return buttonStyles.busy.primary;
+	}
+
+	return buttonStyles.busy.generic;
+};
+
+const getPressedStyles = ( props ) => {
+	if ( props.isPressed ) {
+		return buttonStyles.shared.pressed;
+	}
+
+	return '';
+};
+
+const getSmallStyles = ( props, hasIcon, hasText ) => {
+	if ( props.isSmall ) {
+		if ( hasIcon && ! hasText ) {
+			return buttonStyles.shared.smallOnlyIcon;
+		}
+
+		return buttonStyles.shared.small;
+	}
+
+	return '';
+};
+
+const getIconStyles = ( hasIcon, hasText ) => {
+	if ( hasIcon ) {
+		if ( hasText ) {
+			return buttonStyles.shared.iconWithText;
+		}
+
+		return buttonStyles.shared.icon;
+	}
+
+	return '';
+};
```

Comment:
I don't _love_ this whole slew of functions. Any ideas for how to improve this or different approaches that could be taken to accomplish the same thing would be very welcome.